src/python/foglamp/device/server.py

```diff
@@ -61,61 +82,109 @@ class Server:
+                                                        '{} Device'.format(cls._plugin_name), True)
 
-            config = await configuration_manager.get_category_all_items(category)
+            config = await configuration_manager.get_category_all_items(cls._plugin_name)
 
             try:
-                plugin_module = config['plugin']['value']
+                plugin_module_name = config['plugin']['value']
             except KeyError:
-                _LOGGER.warning("Unable to obtain configuration of module for plugin {}".format(plugin))
+                _LOGGER.warning("Unable to obtain configuration of module for plugin {}".format(cls._plugin_name))
                 raise
 
             try:
-                cls._plugin = __import__("foglamp.device.{}_device".format(plugin_module), fromlist=[''])
+                cls._plugin = __import__("foglamp.device.{}_device".format(plugin_module_name), fromlist=[''])
             except Exception:
-                error = 'Unable to load module {} for device plugin {}'.format(plugin_module,
-                                                                               plugin)
+                error = 'Unable to load module {} for device plugin {}'.format(plugin_module_name,
+                                                                               cls._plugin_name)
                 raise
 
             default_config = cls._plugin.plugin_info()['config']
 
-            await configuration_manager.create_category(category, default_config,
-                                                        '{} Device'.format(plugin))
+            await configuration_manager.create_category(cls._plugin_name, default_config,
+                                                        '{} Device'.format(cls._plugin_name))
 
-            config = await configuration_manager.get_category_all_items(category)
+            config = await configuration_manager.get_category_all_items(cls._plugin_name)
 
             # TODO: Register for config changes
 
             cls._plugin_data = cls._plugin.plugin_init(config)
             cls._plugin.plugin_run(cls._plugin_data)
 
-            await Ingest.start(core_mgt_host, core_mgt_port)
+            await Ingest.start(cls._core_management_host,cls._core_management_port)
         except Exception:
             if error is None:
-                error = 'Failed to initialize plugin {}'.format(plugin)
+                error = 'Failed to initialize plugin {}'.format(cls._plugin_name)
             _LOGGER.exception(error)
             print(error)
             asyncio.ensure_future(cls._stop(loop))
 
     @classmethod
+    def _make_microservice_management_app(cls):
+        # create web server application
+        cls._microservice_management_app = web.Application(middlewares=[middleware.error_middleware])
+        # register supported urls
+        routes.setup(cls._microservice_management_app)
```

Comment:
I'm not sure, why device service routes should point to service registry methods?

`ping` will be device service ping; not core service ping. @MarkRiddoch thoughts?

Reply:
Yes, the ping is to ping the management API and each micro service will have one.

Reply:
I doubt! `http://localhost:<device microservice mgt port>/service/ping` should give info of device microservice or core micro service managment api
Personally  I believe its worth to wait to fix this routes binding